hildon/hildon-live-search.c

```diff
@@ -761,11 +760,13 @@ hildon_live_search_init                         (HildonLiveSearch *self)
 
     priv->entry = hildon_entry_new (HILDON_SIZE_FINGER_HEIGHT);
 
+#ifdef MAEMO_GTK
     /* Unset the autocap and dictionary input flags from the
        HildonEntry. */
-    imode = hildon_gtk_entry_get_input_mode (GTK_ENTRY (priv->entry));
+    HildonGtkInputMode    imode = hildon_gtk_entry_get_input_mode (GTK_ENTRY (priv->entry));
```

Comment:
Can we keep the def of the var at the top, in an ifdef? I think that's better C practice, is it not?

Reply:
I'd disagree it's a better practice. It comes from the times of ANSI C (C90), where you had to place all the variable definitions at the top. Since C99 it's permitted to define them everywhere.

I'd prefer not to have unused variable in my code. It's error prone (it's an uninitialized variable which I can unwittingly pass somewhere else and trigger UB) and will trigger warnings with correct compiler options. I know that libhildon currently has over 700 warnings, but I'd prefer not to add more of them. There's no performance advantage, since the resulting assembly will most likely be the same anyway. This way I'm sure that I won't use this variable unless I use the Maemo Gtk+ API.

Reply:
Done.